src/clojure_tic_tac_toe/minimax.clj

```diff
@@ -23,40 +24,51 @@
     (apply (max-or-min-based-on-player player) scores)))
 
 
-(defn- create-map-of-moves-and-scores
-  [moves scores]
-  (zipmap moves scores))
-
-(defn- get-pairs-with-target-score
-  [moves-and-scores target-score]
-  (filter #(-> % val (= target-score)) moves-and-scores))
-
-(defn- get-move-from-pairs
-  [pairs]
-  (first (first pairs)))
-
-(defn- get-move-based-on-score
-  [moves scores target-score]
-  ((comp get-move-from-pairs
-         #(get-pairs-with-target-score % target-score)
-         create-map-of-moves-and-scores)
-   moves scores))
-
-(defn minimax-move-and-score
-  [game-state {:keys [winning-moves valid-moves] :as initial-data}]
-  (if (:finished? game-state)
-    (let [score (game_handler/calculate-score game-state winning-moves)]
-      {:move :invalid-move, :score score})
-    (let [moves (game_handler/get-available-moves game-state valid-moves)
-          new-game-states (create-new-game-states game-state initial-data moves)
-          moves-and-scores (map #(minimax-move-and-score % initial-data)
-                                new-game-states)
-          scores (get-scores moves-and-scores)
-          score (get-score-based-on-player scores game-state)
-          move (get-move-based-on-score moves scores score)]
-      {:move move, :score score})))
+(defn- create-score-info-list
+  [scores moves total-moves]
+  (map #(assoc {} :score %1 :move %2 :total-moves %3)
+       scores moves total-moves))
+
```

Comment:
I have 2 spaces between each private function that is used in minimax, and only 1 space between private functions that are used in those private functions. So, for example, create-new-game-states is used in minimax, so I have 2 spaces after it. Same thing with get-scores. However, for get-score-based-on-player, it needs max-or-min-based-on-player, so I only have 1 space between max-or-min-based-on-plyaer and get-score-based-on-player. I know this is kind of confusing, because all the functions are private anyway (but I want to keep it as is, to make it easier for me to distinguish which of the private functions is higher level).

Reply:
Got it. I'm thinking maybe a comment separating the private functions might make more sense. It's not really clear from the code itself to the reader what the 2 spaces vs 1 space means unless you explain it to them like you just did to me

Reply:
I added in the comments.